### ISSUE TITLE
Style/RedundantRegexpCharacterClass flags single characters in interpolations

### ISSUE THREAD

**Issue by beechtom** [rubocop-hq/rubocop#8098]
## Expected behavior

The Style/RedundantRegexpCharacterClass should not flag or fix single characters in brackets in interpolations. Interpolations with bracket notation are typically used to access objects and should not be treated as single-element classes.

## Actual behavior

The Style/RedundantRegexpCharacterClass flags and fixes single characters in brackets in interpolations. Interpolations with bracket notation is viewed as a single-element class and has the brackets removed. This results in the Lint/Syntax cop flagging the change due to unrecognized characters in the interpolation. See below.

## Steps to reproduce the problem

Take this trivial script:

```ruby
string = "one.two.three"
regexp = /#{string.split('.')[0]}/

puts regexp.match('one')
```

The expected behavior is that the regexp should resolve to `/one/`. However, running the Style/RedundantRegexpCharacterClass will flag `[0]` as a single-element class even though it is valid notation for accessing the first element of an array. Attempting to auto-correct with this cop results in the following change to the file:

```ruby
string = "one.two.three"
regexp = /#{string.split('.')0}/

puts regexp.match('one')
```

The interpolation `string.split('.')0` has invalid syntax and results in the Lint/Syntax cop flagging the change due to an unexpected integer:

```shell
E: Lint/Syntax: unexpected token tINTEGER
(Using Ruby 2.5 parser; configure using TargetRubyVersion parameter, under AllCops)
      regex = /#{string.split('.')0}/
```

## RuboCop version

```shell
$ rubocop -V
0.85.0 (using Parser 2.7.1.3, rubocop-ast 0.0.3, running on ruby 2.6.3 x86_64-darwin19)
```
